accepted/00097-recurring-actions.md

```diff
@@ -0,0 +1,193 @@
+
+The workflow to add new recurring action types would look like:
+1) Think about which action specific parameters need to be stored to schedule the action from the `RecurringActionApplyJob` `Taskomatic` job.
+2) Add a new element to the `ActionType` enum in `RecurringActionType`.
+3) Add new column(s) to the `suseRecurringActionType` table that store the parameters from  1).
+4) Create a new `Recurring<Action_Type_Name>Type` class extending the base `RecurringActionType` class and include the parameters (including getters/setters) from 1).
+5) Receive action type and parameters from the frontend (add CRUD support on the frontend for the type)
+6) Add support for the action type in `RecurringActionController/Manager`
+7) Make the `RecurringActionApplyJob` schedule the action.
+
+For action type specific backend implementations see below.
+
+### Frontend
+
+The changes required on the frontend part depend on the type of recurring action we want to implement. In general the frontend code used for the highstate implementation can also be reused for creating/listing/modifying and deleting of other recurring action types.
+
+Some terminology changes regarding recurring actions on the frontend (and ideally on the backend to avoid confusion) are going to be required.
+
+To clarify:
+1) All `Recurring States` mentions that do not specifically mean the currently existing recurring `highstate` are going to be renamed to `Recurring Actions`.
+2) All `Recurring States` mentions that related to the `highstate` are going to be renamed to `Recurring Highstate`
+
+The list of `all` recurring actions in the main menu `Schedules` -> `Recurring States (renamed to Recurring Actions)` will be used to display all created recurring action schedules independent of entity (e.g. minion) and action type.
+
+To support this a new column called `Action Type`(Highstate, Custom State, openSCAP, etc.) is going to be added to the list to be able to distinguish between them.
+
+Currently the general list does not indicate for which entity the schedule was created for (only the entity type) this should be changed by adding a title to the entity type displaying the entity name as well as a link to the entity.
+
+Since it not currently possible and the number of recurring action schedules is expected to grow significantly we also want to support being able to filter the list as well as implement server side pagination.
+
+On the system level the `Recurring States` tab is currently displayed as a sub to of the `States` tab. It will be moved to the top level tabs and renamed to `Recurring Actions` since it's no longer going to be state specific.
+
+For action type specific frontend implementations see below.
+
+## Specific implementations
+
+### Custom States (Config Channels)
+
+The idea here is to schedule a `Apply states[custom]` action on a recurring basis. The states executed depend on what is configured in `States` -> `Configuration Channels` on System / Group / Org level.
```

Comment:
I would suggest to allow to add items as "free text".
We have the list of config channels which you can select to add the "string" to a list.
Additionally you can add entries using "free text".
All together is put in a listbox which can be sorted using "arrow buttons" moving items up and down.

Reply:
(I missed that discussion and cannot find "free text" in the RFC, so maybe my comment is obsolete)

I am not sure if I like the idea of the "free text" option. How should customers know what to write, wouldn't it be better to at least scan for available states and list their names for selection?

Further I doubt that all (or even any) of the internally used states should be available for selection. IIRC they often serve a very specific purpose that is not really documented anywhere as they are basically implementation details, and often require special pillar data as input, which means that it would probably not even work to just call them. I might suggest that only a very limited list of internal states that make sense in this context can be selected, like e.g. channels, if at all.